dlls/egon.cpp

```diff
@@ -151,10 +151,15 @@ BOOL CEgon::HasAmmo( void )
 
 void CEgon::UseAmmo( int count )
 {
+#if !CLIENT_DLL
+	if( g_pGameRules->IsBustingGame())
+		return;
+
 	if( m_pPlayer->m_rgAmmo[m_iPrimaryAmmoType] >= count )
 		m_pPlayer->m_rgAmmo[m_iPrimaryAmmoType] -= count;
 	else
 		m_pPlayer->m_rgAmmo[m_iPrimaryAmmoType] = 0;
+#endif
```

Comment:
See code below.

Reply:
I understand but doesn't NOT decreasing ammo in usual game will break prediction, where player would have few more frames where they can use Egon when the ammo is already gone?

Reply:
Well, I see this functions isn't called when CLIENT_DLL is defined.

But my point is instead of keeping it's empty, it's better to just disable the `g_pGameRules` check (as properly simulating game rules on client side is completely different story), just in case somebody calls UseAmmo in their mod.

And yes, client side prediction properly simulate the `m_rgAmmo` array in player entity.